ipie/analysis/blocking.py

```diff
@@ -390,3 +390,41 @@ def analyse_ekt_ipea(filename, ix=None, cutoff=1e-14, screen_factor=1):
     fockT = numpy.dot(X.conj().T, numpy.dot(fock_1p_av, X))
     eea, eea_vec = numpy.linalg.eigh(fockT)
     return (eip, eip_vec), (eea, eea_vec)
+
+
+def block_ratios(num: numpy.ndarray, denom: numpy.ndarray):
+    r"""Block averaging of ratios.
+
+    Parameters
+    ----------
+    num : :class:`numpy.ndarray`
+        Numerator.
+    denom : :class:`numpy.ndarray`
+        Denominator.
+
+    Returns
+    -------
+    mean : :class:`numpy.ndarray`
+        Mean of the ratio.
+    """
+    mean = numpy.mean(num) / numpy.mean(denom)
+    n_samples = num.size
+    print(f"Complex mean: {mean:.8e}")
+    header_0 = "Block size"
+    header_1 = "# of blocks"
+    header_2 = "Mean"
+    header_3 = "Error"
+    print(f" {header_0:>17s} {header_1:>17s} {header_2:>17s} {header_3:>17s}")
+    block_sizes = numpy.array([1, 2, 5, 10, 20, 50, 100, 200, 300, 400, 500, 1000, 10000])
```

Comment:
what if you have more than 10000 blocks? block_sizes = [some_sensible_spacing for i in range(1, whatever_a_sensible_upper_limit_is)]

Reply:
the block_size here does not refer to the number of blocks, anyway, I replaced this with a cleaner jackknife method